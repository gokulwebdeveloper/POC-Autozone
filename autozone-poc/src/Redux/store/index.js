import {createStore, applyMiddleware, compose} from 'redux';
import rootReducer from '../reducers';
import rootSaga from '../middlewares/saga';
import createSagaMiddleware from 'redux-saga';

// create the saga middleware
export const sagaMiddleware = createSagaMiddleware();

//list of middlewares
const middlewares = [sagaMiddleware];

//apply middleware on store
const store = createStore(rootReducer, applyMiddleware(...middlewares));

//run saga generator function
sagaMiddleware.run(rootSaga);

export default store;