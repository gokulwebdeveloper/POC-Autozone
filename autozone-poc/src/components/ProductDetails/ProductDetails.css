.owl-prev {
    position: absolute;
    left: 0;
    left: 20px;
    height: 30px;
    text-align: center;
    transform: translateY(-50%);
    top: 50%;
    background: url(../../assets/img/core-img/long-arrow-left.svg) no-repeat !important;
    width: 30px;
    opacity: 0.4;
    border: none !important;
  }
  
  .owl-nav span {
    visibility: hidden;
  }
  
  button.owl-prev:focus {
    border:transparent 0px
  }
  
  .owl-next {
    position: absolute;
    right: 0;
    right: 20px;
    height: 30px;
    text-align: center;
    transform: translateY(-50%);
    top: 50%;
    background: url(../../assets/img/core-img/long-arrow-right.svg) no-repeat !important;
    width: 30px;
    opacity: 0.4;
  }
    
  button.owl-prev:focus, 
  button.owl-next:focus {
    outline: none;
  }

  .form-control.quantitybox {
    width: 40px;
    height: 35px;
    margin-right: 5px;
    margin-left: 5px;
    text-align: center;
    font-weight: 600;
    font-size: 18px;
  }

  .single_product_desc .quantitybutton{
    min-width: 30px;
    padding: 0px;
    background-color: transparent;
    border: none;
    outline: none;
  }

  .single_product_desc .quantitybutton:focus{
    background-color: transparent;
    border: none;
    outline: none;
  }

  .single_product_details_area .single_product_desc .product-price .new-price{
    display: inline-block;
    font-family: "Ubuntu", sans-serif;
    font-size: 24px;
    color: #dc0345;
    font-weight: 700;
    text-decoration: none;
    margin-right: 0px;
  }

  .owl-stage .owl-item:active{
    padding-left: 50px;
    padding-right: 50px;
  }

  .owl-carousel .owl-stage-outer .owl-stage .owl-item div{
    padding-left: 50px;
    padding-right: 50px;
    padding-top: 20px;
    padding-bottom: 20px;
  }

  .single_product_desc .priceperitem{
  font-size: 10px;
  height: 10px;
  letter-spacing: normal;
  text-transform: none;
  font-weight: normal;
  color: #484848;
  }

  .single_product_desc .outofstock{
    color: #dc0345;
    height: 10px;
    margin: 0px 0px 10px 0px;
  }