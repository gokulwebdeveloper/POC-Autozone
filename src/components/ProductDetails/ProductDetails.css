
  .form-control.quantitybox {
    width: 40px;
    height: 35px;
    margin-right: 5px;
    margin-left: 5px;
    text-align: center;
    font-weight: 600;
    font-size: 18px;
  }

  .single_product_desc .quantitybutton{
    min-width: 30px;
    padding: 0px;
    background-color: transparent;
    border: none;
    outline: none;
  }

  .single_product_desc .quantitybutton:focus{
    background-color: transparent;
    border: none;
    outline: none;
  }

  .single_product_details_area .single_product_desc .product-price .new-price{
    display: inline-block;
    font-family: "Ubuntu", sans-serif;
    font-size: 24px;
    color: #dc0345;
    font-weight: 700;
    text-decoration: none;
    margin-right: 0px;
  }

  .single_product_desc .priceperitem{
  font-size: 10px;
  height: 10px;
  margin: 10px;
  letter-spacing: normal;
  text-transform: none;
  font-weight: normal;
  color: #484848;
  }

  .single_product_desc .outofstock{
    color: #dc0345;
    height: 10px;
  }

.error-template {padding: 40px 15px;text-align: center;}
.error-actions {margin-top:15px;margin-bottom:15px;}
.error-actions .btn { margin-right:10px; }

.single_product_desc .rating-row{
    display: inline-flex;
    margin-bottom: -50px;
}

.single_product_desc .rating-row.warranty{
  color: #484848;
  margin-right: 10px;
  margin-left: 10px;
}

.single_product_desc .rating-row .rating-number{
  color: #484848;
  margin-right: 10px;
  margin-left: 10px;
}

.orderByMessage{
  color: #484848;
  margin-bottom: 10px;
}

.single_product_thumb .slick-slider button.slick-arrow.slick-prev,
.single_product_thumb .slick-slider button.slick-arrow.slick-next{
  margin-left: 10px;
  margin-right: 10px;
  z-index: 100;
  background-color: lightgray;
}

.single_product_thumb .slick-slider button.slick-arrow.slick-prev:hover, 
.single_product_thumb .slick-slider button.slick-arrow.slick-prev:focus,
.single_product_thumb .slick-slider button.slick-arrow.slick-next:hover, 
.single_product_thumb .slick-slider button.slick-arrow.slick-next:focus {
  outline: none;
  background-color: lightgray;
}

.single_product_details_area .single_product_thumb, .single_product_details_area .single_product_desc{
  padding: 10px;
}

.single_product_details_area {
  margin-left: 10px;
  margin-right: 10px;
}

.single_product_details_area .single_product_thumb .slick-list{
  margin-bottom: 15px;
}