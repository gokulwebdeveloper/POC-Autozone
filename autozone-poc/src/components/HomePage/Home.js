import React from 'react';
import BgImg from '../../utlis/img/azo-product-img/homePageImages/autozone-banner2.jpg';
import BgImgCatArea from '../../utlis/img/azo-product-img/homePageImages/step-brake.jpg';
import PropTypes from 'prop-types';
import Button from '../../utlis/button/Button';
import TopCatagoryArea from '../../utlis/top_catagory_area/TopCatagoryArea';
import SimpleSlider from '../../utlis/slick_slider/SimpleSlider';
import ScrollUpArrowBtn from '../../utlis/button/ScrollUpArrowBtn';
import BrandArea from './Brands/BrandArea';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { addToCart, getProductDetail } from '../../Redux/actions';


const Home = props => { 
  let topCatagoryData = [];
  let currentProduct;
  props.catagoryData ? props.catagoryData.map((topCatagory, i) => {
    currentProduct = topCatagory;
    if(topCatagory.offer_badge == "New") {
      topCatagoryData.push(topCatagory);
    }
  }) : ''
  const bgImgStyle = {
    'backgroundImage': `url(${BgImg})`
  }
  
  const bgImgCatArea = {
    'backgroundImage': `url(${BgImgCatArea})`
  }

  let customeSettings = {
    dots: true,
    infinite: true,
    speed: 1000,
    slidesToShow: 4,
    slidesToScroll: 1,
    autoplay: true,
    autoplaySpeed: 2000
  };

  function addToCart()
    {
        let qty = 1;
        let price = parseFloat(currentProduct.offer_price);
        let total = parseFloat(qty * price);
        addKeyValue(currentProduct,"quantity", qty);
        addKeyValue(currentProduct,"price", total);
        props.dispatchAddToCart(currentProduct);
    }

    function addKeyValue(obj, key, data){
        obj[key] = data;
      }

    return (
      <div>
        <ScrollUpArrowBtn/>
        <section className="welcome_area bg-img" style={bgImgStyle}>
          <div className="container h-140">
            <div className="row h-100 align-items-center">
              <div className="col-12">
                <div className="hero-content">
                <Link  to={{
                        pathname:'/shops',
                        filter: {
                        filterType: "Accessories",
                        filter: 'All'
                        }
                      }}><Button btnText="view collection" /></Link >
                </div>
              </div>
            </div>
          </div>
        </section>
        <div className="top_catagory_area section-padding-80 clearfix">
          <div className="container">
            <div className="row justify-content-center">{
               topCatagoryData ? topCatagoryData.map((catagoryValues, i) => {
                return (
                  <TopCatagoryArea
                    key={i}
                    topCatagoryAreaText={catagoryValues.category}
                    bgStyleImg={catagoryValues.product_image[0]}
                  />
                )
              }) : ''
            }
            </div>
          </div>
        </div>
        <div className="cta-area">
          <div className="container">
            <div className="row">
              <div className="col-12">
                <div className="cta-content background-overlay bg-img" style={bgImgCatArea}>
                  <div className="h-140 d-flex align-items-center justify-content-end">
                    <div className="cta--text">
                    <Button btnText="Buy Now" onClick={()=>props.getProductDetail(currentProduct.id), addToCart}/>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <section className="new_arrivals_area section-padding-80 clearfix">
          <div className="container">
            <div className="row">
              <div className="col-12">
                <div className="section-heading text-center">
                  <h2>Popular Products</h2>
                </div>
              </div>
            </div>
          </div>
          <div className="container">
            <div className="row">
              <div className="col-12">
                <div className="popular-products-slides owl-carousel owl-theme owl-loaded">
                  <SimpleSlider settings={customeSettings}/>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section>
          <BrandArea />
        </section>
      </div>
    );
};
const mapStateToProps = productData => {
  return {
    catagoryData: productData.productData.data
  };
}

const mapDispatchToProps = (dispatch) =>{
  return {
      dispatchAddToCart: product => dispatch(addToCart(product)),
      getProductDetail: id => dispatch(getProductDetail(id))
  }
} 

Home.propTypes = {
  topCatagoryAreaText: PropTypes.string
};

Home.defaultProps = {
  topCatagoryAreaText: '',
  customeSettings: {
    dots: true,
    infinite: true,
    speed: 1000,
    slidesToShow: 4,
    slidesToScroll: 1,
    autoplay: true,
    autoplaySpeed: 2000
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);