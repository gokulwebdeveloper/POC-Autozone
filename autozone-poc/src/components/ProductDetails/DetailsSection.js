import React, {useEffect} from 'react';
import '../../matchMedia.mock';
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import '../ProductDetails/ProductDetails.css';
import Button from '../../utlis/button/Button';
import PropTypes from 'prop-types';
import Rating from 'react-rating';
import { connect } from 'react-redux';
import {addToCart} from '../../Redux/actions';
import moment from 'moment';
import add from "../../assets/img/core-img/add.svg"
import minus from "../../assets/img/core-img/minus.svg"

const DetailsSection = ({currentProduct, dispatchAddToCart}) => {
    const nextDay = new Date();
    const today = new Date();
    nextDay.setDate(nextDay.getDate() + 1);

    const deliveryMessage = "Order Before 9:00 PM "+moment(today).format('MMM DD')+" for "+moment(nextDay).format('MMM DD')+" delivery";

    useEffect( ()=>{
        
        updateFavoriteIco(currentProduct);
        let txtcartquantity = document.getElementById("txtcartquantity");

        if(txtcartquantity)
        {
            if(currentProduct.product_instock)
                currentProduct.quantity ? txtcartquantity.value = currentProduct.quantity : txtcartquantity.value = 1;
            else
                txtcartquantity.value = 0;
        }

        let outOfStock = document.getElementById("outOfStock");
        if(outOfStock)
        {
            if(currentProduct.product_instock)
                outOfStock.style.visibility = "hidden";
            else
                outOfStock.style.visibility = "visible";
        }
     });

    return (
        <section className="single_product_details_area d-flex align-items-center">
            <div className="single_product_thumb clearfix">
            <Slider 
                    dots={true}
                    infinite={true}
                    speed={1000}
                    slidesToShow={1}
                    slidesToScroll={1}
                    swipeToSlide={true}
                    initialSlide={0}
                    autoplay={true}
                    autoplaySpeed={4000}
                    arrows={true}
                    >

                    {
                    currentProduct.product_image ? 

                    currentProduct.product_image.map((image, index) => {
                    return (
                        <img src={image} alt={currentProduct.product_name} key={index}/>
                        )
                    })
                    
                    : "No Images"
                    }
                </Slider>
            </div>

            {/* <!-- Single Product Description --> */}
            <div className="single_product_desc clearfix">
                <span>{currentProduct.product_brand}</span>
                    <h2>{currentProduct.product_name}</h2>
                    <p className="rating-row">
                         <Rating
                    initialRating={currentProduct.rating}
                    readonly
                    className="rating"
                    emptySymbol="fa fa-star-o fa-2x" 
                    fullSymbol="fa fa-star fa-2x"
                    fractions={2}
                    />
                    <span className="rating-number">({currentProduct.rating})</span>
                    <span className="warranty">{currentProduct.warranty}</span>
                    </p>

                    <h6 id="outOfStock" className="outofstock">Out of stock</h6>
                   
                <p className="product-desc">{currentProduct.product_desc}</p>
            
                    <p className="product-price"><span className="old-price">${currentProduct.original_price}</span>
                    <span className="new-price"> $</span>
                        <span id="cartprice" className="new-price">{currentProduct.offer_price}</span></p>
                
                <div id="divQty" className="d-flex">
                <div className="d-flex">
                    <button id="btnDecreaseQty" className="quantitybutton"
                            onClick={()=>onDecreaseQty(currentProduct)} 
                            disabled={currentProduct.product_instock === true ? false : true}>
                        <img src={minus} alt="decrease button"/>
                    </button>
                </div>
                <div className="d-flex">
                    <input id="txtcartquantity" className="form-control quantitybox"
                    onChange={()=>onQtyChange(currentProduct)} onBlur={()=>onQtyLostFocus(currentProduct)} maxLength="1"
                    disabled={currentProduct.product_instock === true ? false : true}/>
                </div>
                <div className="d-flex">
                    <button className="quantitybutton"
                    onClick={()=>onIncreaseQty(currentProduct)}
                    disabled={currentProduct.product_instock === true ? false : true}>
                        <img src={add} alt="increase button"/>
                    </button>
                </div>
                </div>
                <div id="pricePerItem" className="priceperitem" ></div>
                
            <div className="orderByMessage">{deliveryMessage}</div>
                    {/* <!-- Cart & Favourite Box --> */}
                    <div className="cart-fav-box d-flex align-items-center">
                        <Button id="btnAddtoCart" btnText="Add to Cart" 
                        disabled={currentProduct.product_instock === true ? false : true}
                        onClick={()=>addToCartFn(currentProduct, dispatchAddToCart)}/>
                        {/* <!-- Favourite --> */}
                        <div className="product-favourite ml-4" onClick={changeFavStatus}>
                            <a id="favme" className="favme fa fa-heart"></a>
                        </div>
                    </div>
                    
            </div>
        </section>
    );
};

const maxAllowedQty = 5;

function addToCartFn(currentProduct, dispatchAddToCart)
{
    let txtcartquantity = document.getElementById("txtcartquantity");

    let qty = parseInt(txtcartquantity.value);

    addKeyValue(currentProduct,"quantity", qty);
    addKeyValue(currentProduct,"price", currentProduct.offer_price);

    dispatchAddToCart(currentProduct);
}

function updateFavoriteIco(currentProduct){
let currentCntrl = document.getElementById("favme");

if(currentCntrl!=null)
{
    if(currentProduct.isFavourite)
        currentCntrl.className = "favme fa fa-heart active";
    else
        currentCntrl.className = "favme fa fa-heart";
}
}

function changeFavStatus() {
let currentCntrl = document.getElementById("favme");

if(currentCntrl==null)
    return;

if(!currentCntrl.className.includes("active"))
    currentCntrl.className = "favme fa fa-heart active";
else
    currentCntrl.className = "favme fa fa-heart";
}

function onQtyChange(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

let qty = parseInt(currentCntrl.value);

if(currentCntrl.value == null)
{
    currentCntrl.value = 1;
    updateEach(qty, currentProduct);
}
else if(qty > maxAllowedQty)
{
    currentCntrl.value = 1;
    updateEach(qty, currentProduct);
}
else if(qty > 1 && qty <= maxAllowedQty)
{
    updateCartPrice(currentProduct);
    updateEach(qty, currentProduct);
}
else
    return;
}

function onQtyLostFocus(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

if(currentCntrl.value == null || currentCntrl.value == "")
{
    currentCntrl.value = 1;
    updateCartPrice(currentProduct);  
}
}

function onIncreaseQty(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

if(currentCntrl.value == null || currentCntrl.value == "")
    currentCntrl.value = 1;
else
{
    let val = parseInt(currentCntrl.value);
    
    if(val < maxAllowedQty)
    {
        currentCntrl.value = val + 1;
        updateCartPrice(currentProduct);
    }
    else
        return;
}
}

function onDecreaseQty(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

if(currentCntrl.value == null || currentCntrl.value == "")
    currentCntrl.value = 1;
else
{
    let val = currentCntrl.value;
    if(val > 1)
    {
        currentCntrl.value = val - 1;
        updateCartPrice(currentProduct);
    }
    else
        return;
}
}

function updateCartPrice(currentProduct)
{
let currentCntrl = document.getElementById("txtcartquantity");

if(currentCntrl==null)
    return;

let cartCntrl = document.getElementById("cartprice");

if(cartCntrl==null)
    return;

let qty = parseInt(currentCntrl.value);
let price = parseFloat(currentProduct.offer_price);
let total = qty * price;
cartCntrl.innerHTML = total;
updateEach(qty, currentProduct);
}

function updateEach(qty, currentProduct)
{
let pricePerItem = document.getElementById("pricePerItem");

if(pricePerItem==null)
    return;
        
if(qty == 1)
    pricePerItem.innerHTML = "";
else
    pricePerItem.innerHTML = "Each $ "+currentProduct.offer_price;
}

function addKeyValue(obj, key, data){
obj[key] = data;
}

DetailsSection.propTypes = {
    currentProduct: PropTypes.object,
    dispatchAddToCart: PropTypes.func
}

const mapStateToProps = productDetail =>{
    return productDetail;
}

const mapDispatchToProps = (dispatch) =>{
    return {
        dispatchAddToCart: product => dispatch(addToCart(product))
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(DetailsSection);