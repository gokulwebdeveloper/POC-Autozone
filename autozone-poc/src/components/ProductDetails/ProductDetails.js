import React from "react";
import '../../requiredLoader'
import 'owl.carousel/dist/assets/owl.carousel.css';
import OwlCarousel from 'react-owl-carousel';
import '../ProductDetails/ProductDetails.css';
import Button from '../../utlis/button/Button'

const ProductDetails = () => {
    var styleDisplayNone = {
        display:'none',
     };

     var styleowlstage = {
        transform: 'translate3d(-2295px, 0px, 0px)',
         transition: 'all 1s ease 0s',
         width: 3213
    };

    var styleowlitems = {
       width: 459,
    };

    function changeFavStatus() {
        var currentCntrl = document.getElementById("favme");

        if(currentCntrl==null)
            return;

        if(!currentCntrl.className.includes("active"))
            currentCntrl.className = "favme fa fa-heart active";
        else
            currentCntrl.className = "favme fa fa-heart";
    }

    return (
        <section className="single_product_details_area d-flex align-items-center">
            <div className="single_product_thumb clearfix">
            <OwlCarousel
     className={'owl-carousel'}
     loop={true}
     nav={true}
     dots={false}
     autoplay={true}
     autoplayTimeout={5000}
     autoplaySpeed={1000}
     fluidSpeed={1000}
     items={1}
     responsiveRefreshRate={200}
>

<div><img src="src/utlis/img/product-img/product-big-1.jpg" alt=""/></div>
<div><img src="src/utlis/img/product-img/product-big-2.jpg" alt=""/></div>
<div><img src="src/utlis/img/product-img/product-big-3.jpg" alt=""/></div>
</OwlCarousel>
</div>

        {/* <!-- Single Product Description --> */}
        <div className="single_product_desc clearfix">
            <span>mango</span>
            <a href="cart.html">
                <h2>One Shoulder Glitter Midi Dress</h2>
            </a>
            <p className="product-price"><span className="old-price">$65.00</span> $49.00</p>
            <p className="product-desc">Mauris viverra cursus ante laoreet eleifend. Donec vel fringilla ante. Aenean finibus velit id urna vehicula, nec maximus est sollicitudin.</p>

            {/* <!-- Form --> */}
            <form className="cart-form clearfix" method="post">

                {/* <!-- Select Box --> */}
                <div className="select-box d-flex mt-50 mb-30">
                    <select id="productSize" className="nice-select mr-5">
                        <option >Size: XL</option>
                        <option >Size: X</option>
                        <option >Size: M</option>
                        <option >Size: S</option>
                    </select>
                    <select id="productColor" className="nice-select mr-5">
                        <option >Color: Black</option>
                        <option >Color: White</option>
                        <option >Color: Red</option>
                        <option >Color: Purple</option>
                    </select>
                </div>
                {/* <!-- Cart & Favourite Box --> */}
                <div className="cart-fav-box d-flex align-items-center">
                    <Button btnText="Add to Cart"/>
                    {/* <!-- Favourite --> */}
                    <div className="product-favourite ml-4" onClick={changeFavStatus}>
                        <a id="favme" className="favme fa fa-heart"></a>
                    </div>
                </div>
        </form>
        </div>
        </section>
    );
};

export default ProductDetails;