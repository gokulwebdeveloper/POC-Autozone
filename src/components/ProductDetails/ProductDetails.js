import React from 'react';
import DetailsSection from './DetailsSection';
import ProductNotFound from './ProductNotFound';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {getProductDetail} from '../../Redux/actions';

const ProductDetails = (props) => {
    const prodId = props.match?props.match.params.id:null;

    if(prodId){
        if(props.productDetail)
            return <DetailsSection currentProduct={props.productDetail.data[0]} />
        else
            return <ProductNotFound/>
    }
    else
       return <ProductNotFound/>
};

ProductDetails.propTypes = {
    props: PropTypes.object,
}

const mapStateToProps = ({productDetail}) =>{
    return {productDetail};
}

const mapDispatchToProps = {getProductDetail}

export default connect(mapStateToProps, mapDispatchToProps)(ProductDetails);