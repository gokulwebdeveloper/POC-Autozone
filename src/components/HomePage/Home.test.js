import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '../../matchMedia.mock' // Must be imported before the tested file
import Home from './Home';
import { Provider } from 'react-redux';
import store from '../../Redux/store';
import ProductDetails from '../ProductDetails/ProductDetails';
import { BrowserRouter as Route } from 'react-router-dom';

configure({ adapter: new Adapter() });

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
}); 

const mockData = {
    "id": 1,
    "product_image": [
        "../../../utlis/img/azo-product-img/product-1/1.jpg",
        "../../../utlis/img/azo-product-img/product-1/2.jpg",
        "../../../utlis/img/azo-product-img/product-1/3.jpg",
        "../../../utlis/img/azo-product-img/product-1/4.jpg"
    ],
    "product_name": "Duralast Brake Rotor 42251DL",
    "product_brand": "Duralast",
    "original_price": "75.00",
    "offer_price": "55.00",
    "offer_badge": "-30%",
    "category": "Brakes and Traction Control",
    "sub_category": "Brake Rotor",
    "product_desc": "Rotor only, *Coated on hat and edge for rust prevention *Replace in pairs for best performance",
    "product_instock": true,
    "orderByMessage": "Order Before 9:00 PM Jul 01 for Jul 02 delivery",
    "warrantyType": "AutoZone",
    "warranty": "2 Year Warranty",
    "rating": 4.5,
    "isFavourite": false
  };

describe('Home page testing', () => {
    const wrapper = mount(
        <Provider store={store}>
            <Route path="/single-product-details/:id" component={ProductDetails}>
            <Home productData={mockData} />
            </Route>
        </Provider>);
        it("Renders without crashing", ()=> {
            mount(<Provider store={store}>
                <Route path="/single-product-details/:id" component={ProductDetails}>
                <Home productData={mockData} />
                </Route>
            </Provider>);
        });
    
    //test if home component exists
    it("Home Exists", () => {
        expect(wrapper.exists()).toBeTruthy();
    });
    // length of h2 tags
    test('renders the h2 tag called', () => {
        const h2 = wrapper.find("h2");
        expect(h2).toHaveLength(1);
    });

    it("render the SimpleSlider", () => {
        const simpleSliderTag =  wrapper.find('SimpleSlider');
        expect(simpleSliderTag).toHaveLength(1);
    });


    //if it has 1 Button componenent
    it("finding the Button componenent length", () => {
        expect(wrapper.find("Button")).toHaveLength(2);
    });
})