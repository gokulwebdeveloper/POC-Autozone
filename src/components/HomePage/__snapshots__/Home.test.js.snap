// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page testing SnapShot Match 1`] = `
<div>
  <aside
    aria-label="Scroll to top of page"
    className="ScrollUpButton__Container "
    data-testid="react-scroll-up-button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "WebkitTransition": "all 0.5s ease-in-out",
        "backgroundColor": "rgba(50, 50, 50, 0.5)",
        "bottom": 20,
        "cursor": "pointer",
        "height": 50,
        "opacity": 0,
        "position": "fixed",
        "right": -50,
        "transition": "all 0.5s ease-in-out",
        "transitionProperty": "opacity, right",
        "width": 50,
        "zIndex": 1000,
      }
    }
    tabIndex={-1}
  >
    <svg
      style={
        Object {
          "display": "inline-block",
          "fill": "white",
          "height": "100%",
          "stroke": "white",
          "strokeWidth": 0,
          "width": "100%",
        }
      }
      version="1.1"
      viewBox="0 0 32 32"
      x="0"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      y="0"
    >
      <path
        d="M19.196 23.429q0 0.232-0.179 0.411l-0.893 0.893q-0.179 0.179-0.411 0.179t-0.411-0.179l-7.018-7.018-7.018 7.018q-0.179 0.179-0.411 0.179t-0.411-0.179l-0.893-0.893q-0.179-0.179-0.179-0.411t0.179-0.411l8.321-8.321q0.179-0.179 0.411-0.179t0.411 0.179l8.321 8.321q0.179 0.179 0.179 0.411zM19.196 16.571q0 0.232-0.179 0.411l-0.893 0.893q-0.179 0.179-0.411 0.179t-0.411-0.179l-7.018-7.018-7.018 7.018q-0.179 0.179-0.411 0.179t-0.411-0.179l-0.893-0.893q-0.179-0.179-0.179-0.411t0.179-0.411l8.321-8.321q0.179-0.179 0.411-0.179t0.411 0.179l8.321 8.321q0.179 0.179 0.179 0.411z"
        transform="scale(1.4) translate(1,-5)"
      />
    </svg>
  </aside>
</div>
`;
